spotipy/oauth2.py

```diff
@@ -272,33 +286,26 @@ def get_cached_token(self):
         """
         token_info = None
 
-        if not self.cache_path and self.username:
-            self.cache_path = ".cache-" + str(self.username)
-        elif not self.cache_path and not self.username:
-            raise SpotifyOauthError(
-                "You must either set a cache_path or a username."
-            )
+        try:
+            f = open(self.cache_path)
+            token_info_string = f.read()
+            f.close()
+            token_info = json.loads(token_info_string)
 
-        if self.cache_path:
-            try:
-                f = open(self.cache_path)
-                token_info_string = f.read()
-                f.close()
-                token_info = json.loads(token_info_string)
+            # if scopes don't match, then bail
+            if "scope" not in token_info or not self._is_scope_subset(
+                self.scope, token_info["scope"]
+            ):
+                return None
 
-                # if scopes don't match, then bail
-                if "scope" not in token_info or not self._is_scope_subset(
-                    self.scope, token_info["scope"]
-                ):
-                    return None
+            if self.is_token_expired(token_info):
+                token_info = self.refresh_access_token(
+                    token_info["refresh_token"]
+                )
 
-                if self.is_token_expired(token_info):
-                    token_info = self.refresh_access_token(
-                        token_info["refresh_token"]
-                    )
+        except IOError:
+            pass
```

Comment:
I know you simply fixed up these try/catch statements. Could we throw in a logger warning similar to line 1026?
```suggestion
            logger.warning("Couldn't read cache at: %s", self.cache_path)
```

Reply:
Yep

Reply:
I ended up doing something similar locally, I like the fix

Reply:
Thanks, I just released it in 2.15.0